packages/components/rating/src/Rating.test.tsx

```diff
@@ -1,12 +1,127 @@
+import { fireEvent } from '@testing-library/dom'
 import { render, screen } from '@testing-library/react'
-import { describe, expect, it } from 'vitest'
+import userEvent from '@testing-library/user-event'
+import { useState } from 'react'
+import { describe, expect, it, vi } from 'vitest'
 
 import { Rating } from './Rating'
 
+const utils = {
+  getInput: () =>
+    screen.getByRole('slider', {
+      name: /rating/i,
+    }),
+  getStars: (container: HTMLElement) =>
+    Array.from(container.querySelectorAll('[data-part="star"]')),
+}
+
 describe('Rating', () => {
   it('should render', () => {
-    render(<Rating />)
+    const { container } = render(<Rating defaultValue={2} />)
+
+    expect(container.querySelector('[data-spark-component=rating]')).toBeInTheDocument()
+    expect(container.querySelectorAll('[data-part="star"]').length).toBe(5)
+  })
+
+  it('should be possible to interact', async () => {
+    const user = userEvent.setup()
+
+    const { container } = render(<Rating size="lg" defaultValue={1} />)
+
+    const stars = utils.getStars(container)
+    const input = utils.getInput()
+    const firstStar = stars.at(0)
+    const thirdStar = stars.at(2)
+
+    if (!thirdStar || !firstStar) throw new Error('Stars element not found')
+
+    // initial value
+    expect(input).toHaveValue('1')
+
+    await user.click(thirdStar)
+    expect(input).toHaveValue('3')
+
+    await user.click(firstStar)
+    expect(input).toHaveValue('1')
+
+    // triggering a change event by directly interacting
+    // with the HTML input (using the keyboard for instance)
+    fireEvent.change(input, { target: { value: '5' } })
```

Comment:
Can't you achieve the same goal with `user-event`?

Reply:
Nope, as we saw yesterday 😢
https://github.com/testing-library/user-event/issues/871

Reply:
The purpose of this test is to make sure that we correctly linked the native `onChange` event with our internal React state. This ensures that when the user interacts with the component, the state is updated accordingly.